snapshot/bazel/fs_util_test.go

```diff
@@ -216,3 +219,42 @@ func TestMaterializeEmptyDir(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// test cancellation functionality
+func TestCancelOperation(t *testing.T) {
+	_, err := tmpTest.TempDir("")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bf := makeTestingFiler()
+	// Override bf's bzCommand's execer with a sim execer
+	bc := bf.tree.(*bzCommand)
+	bc.execer = execers.NewSimExecer()
+
+	cmd := execer.Command{
+		Argv: []string{"pause", "complete 0"},
+	}
+	doneCh := make(chan execer.ProcessStatus)
+
+	go func() {
+		doneCh <- bc.exec(cmd)
+	}()
+
+	// simplest way to ensure cancel gets run after exec actually starts
+	go func() {
+		for {
```

Comment:
do we need any sort of cleanup to ensure this goroutine gets garbage collected?

Reply:
the test completes normally, so not really, although it does look a little clunky.